src/NuGet.Core/NuGet.PackageManagement/Resolution/ResolverGather.cs

```diff
@@ -595,7 +595,7 @@ private async Task InitializeResourcesAsync(CancellationToken token)
                     {
                         var resource = await depResources[source];
 
-                        if (source != null && !_primaryResources.Any(sourceResource => sourceResource.Source.PackageSource.Equals(source)))
+                        if (!_primaryResources.Any(sourceResource => sourceResource.Source.PackageSource.Equals(source)))
```

Comment:
On line 594, the code does a `source.PackageSource`, de-referencing `source`. So it can't be `null` at this point.

Reply:
This raises the question... is the comment above accurate that it can really be `null`?

Reply:
Maybe the comment really means the `resource` variable and the conditional is supposed to be `resource != null && ...` ?

Reply:
Perhaps 🤷‍♂️... I think your refactoring is fine on its own.